### PR TITLE
update Depth Meter for world height changes

### PR DESCRIPTION
**Pull request by pixelfac**
Depth Meter uses **player coords + configurable offsets** instead of layer + section data to calculate the virtual depth of the player in the Abyss. The Depth reading is given in minecraft blocks, not corresponding to canon depth.

### CODE REVIEW

**Comment by pixelfac** [comment]
Can you explain to me a bit about WorldGuard? Like what it does/can do and why we want to depend on it?

Also, are you saying 'dont fix the depth meter until we fix WorldGuard' or 'you can fix it now, but I'd like to change it to depend on WorldGuard in the future'

**Comment by Boy0000** [comment]
image
if this makes any sense lol

Issue with depth meter rn is that DW defines a section as the cube you are in.
With WG regions we can split that cube into more than one section, meaning depth meter will show accurate layer info and depth (still canon but thats cuz MiA config says that it should)

**Comment by pixelfac** [comment]
build is down so I can't easily check when the actual transition is between Edge and Forest, so lets say that its at a depth of 1000. 
```yaml
name: §x§D§B§4§0§4§0Edge of the Abyss
    sub: §x§B§A§1§E§1§E0-1350 meters
    deathMessage: at the §x§D§B§4§0§4§0Edge of the Abyss§f :redw:§f
    depth:
      start: 0
      end: 1000
```
In AbyssLocationAction.kt, `layer = player.location.section.layer` . If I just change the start and end depths in the config, will `player.location.section.layer` correctly give the right layer, or do we need to update how each section knows what layer(s) it's on as well (with WG or whatever)

**Comment by Boy0000** [comment]
If i got dero right, all it would do is let us specify sections accurately and therefore layers aswell. Meaning MiA should correctly display your location, be that canon or block depth

**Comment by Boy0000** [comment]
tldr; since im shit at explaining lol. When DW uses WG regions instead of the system right now to define sections, it should all more or less be accurate. Right now, afaik, the only issues is with the depth not displaying correctly because you are over the threshold of said layer, therefore causing the jumps

**Comment by pixelfac** [comment]
naruhodo. So I'm waiting on worldguard regions

**Comment by Boy0000** [comment]
Yes, will probably still be issues with the old math for calculating etc. But hold of until then
